setup.py

```diff
@@ -9,143 +9,44 @@
-    warnings.warn(f"extension build skipped: {e}")
-finally:
-    if not RUN_BUILD:
-        BUILD_CPP = BUILD_CUDA = False
-        print("Please set environment variable `BUILD_MONAI=1` to enable Cpp/CUDA extension build.")
-    print(f"BUILD_MONAI_CPP={BUILD_CPP}, BUILD_MONAI_CUDA={BUILD_CUDA}, TORCH_VERSION={TORCH_VERSION}.")
-
-
-def torch_parallel_backend():
+# Define extension modules
+ext_modules = []
+if BUILD_MONAI:
     try:
-        match = re.search("^ATen parallel backend: (?P<backend>.*)$", torch._C._parallel_info(), re.MULTILINE)
-        if match is None:
-            return None
-        backend = match.group("backend")
-        if backend == "OpenMP":
-            return "AT_PARALLEL_OPENMP"
-        if backend == "native thread pool":
-            return "AT_PARALLEL_NATIVE"
-        if backend == "native thread pool and TBB":
-            return "AT_PARALLEL_NATIVE_TBB"
-    except (NameError, AttributeError):  # no torch or no binaries
-        warnings.warn("Could not determine torch parallel_info.")
-    return None
-
-
-def omp_flags():
-    if sys.platform == "win32":
-        return ["/openmp"]
-    if sys.platform == "darwin":
-        # https://stackoverflow.com/questions/37362414/
-        # return ["-fopenmp=libiomp5"]
-        return []
-    return ["-fopenmp"]
-
-
-def get_extensions():
-    this_dir = os.path.dirname(os.path.abspath(__file__))
-    ext_dir = os.path.join(this_dir, "monai", "csrc")
-    include_dirs = [ext_dir]
-
-    source_cpu = glob.glob(os.path.join(ext_dir, "**", "*.cpp"), recursive=True)
-    source_cuda = glob.glob(os.path.join(ext_dir, "**", "*.cu"), recursive=True)
-
-    extension = None
-    define_macros = [(f"{torch_parallel_backend()}", 1), ("MONAI_TORCH_VERSION", TORCH_VERSION)]
-    extra_compile_args = {}
-    extra_link_args = []
-    sources = source_cpu
-    if BUILD_CPP:
-        extension = CppExtension
-        extra_compile_args.setdefault("cxx", [])
-        if torch_parallel_backend() == "AT_PARALLEL_OPENMP":
-            extra_compile_args["cxx"] += omp_flags()
-        extra_link_args = omp_flags()
-    if BUILD_CUDA:
-        extension = CUDAExtension
-        sources += source_cuda
-        define_macros += [("WITH_CUDA", None)]
-        extra_compile_args = {"cxx": [], "nvcc": []}
-        if torch_parallel_backend() == "AT_PARALLEL_OPENMP":
-            extra_compile_args["cxx"] += omp_flags()
-    if extension is None or not sources:
-        return []  # compile nothing
-
-    ext_modules = [
-        extension(
-            name="monai._C",
-            sources=sources,
-            include_dirs=include_dirs,
-            define_macros=define_macros,
-            extra_compile_args=extra_compile_args,
-            extra_link_args=extra_link_args,
+        import torch
+        from torch.utils.cpp_extension import CUDAExtension
+
+        print(f"Setup with torch {torch.__version__}")
+
+        ext_modules.append(
+            CUDAExtension(
```

Comment:
As mentioned above, if CUDA isn't present this will fail. We do want to be able to build without CUDA to support environments without acceleration but which can still reasonably do inference, eg. macOS.